tests/unit/api/c/source/test_source.cpp

```diff
@@ -57,15 +72,35 @@
     EXPECT_NE(sacn_lock_fake.call_count, previous_lock_count);                \
     EXPECT_EQ(sacn_lock_fake.call_count, sacn_unlock_fake.call_count);        \
   } while (0)
+#define VERIFY_NO_LOCKING_AND_RETURN_VALUE(function_call, expected_return_value) \
+  do                                                                             \
+  {                                                                              \
+    unsigned int previous_lock_count = sacn_lock_fake.call_count;                \
+    EXPECT_EQ(function_call, expected_return_value);                             \
+    EXPECT_EQ(sacn_lock_fake.call_count, previous_lock_count);                   \
+    EXPECT_EQ(sacn_lock_fake.call_count, sacn_unlock_fake.call_count);           \
+  } while (0)
 
 static const etcpal::Uuid kTestLocalCid = etcpal::Uuid::FromString("5103d586-44bf-46df-8c5a-e690f3dd6e22");
 static const std::string kTestLocalName = std::string("Test Source");
 static const std::string kTestLocalName2 = std::string("Test Source 2");
+static const std::string kTestLocalNameTooLong =
+    std::string("Test Source Name Too Long Test Source Name Too Long Test Source N");
 static const EtcPalIpAddr kTestRemoteAddrV4 = etcpal::IpAddr::FromString("10.101.1.1").get();
 static const EtcPalIpAddr kTestRemoteAddrV6 = etcpal::IpAddr::FromString("2001:db8::1234:5678").get();
-static const sacn_source_t kTestHandle = 123u;
+static const EtcPalIpAddr kTestRemoteAddrs[NUM_TEST_ADDRS] = {
+    etcpal::IpAddr::FromString("10.101.1.1").get(), etcpal::IpAddr::FromString("10.101.1.2").get(),
+    etcpal::IpAddr::FromString("10.101.1.3").get(), etcpal::IpAddr::FromString("10.101.1.4").get()};
+static const EtcPalIpAddr kTestRemoteAddrsWithInvalid[NUM_TEST_ADDRS] = {
+    etcpal::IpAddr::FromString("10.101.1.1").get(), etcpal::IpAddr::FromString("10.101.1.2").get(),
+    etcpal::IpAddr().get(), etcpal::IpAddr::FromString("10.101.1.4").get()};
+static const sacn_source_t kTestHandle = 123;
+static const sacn_source_t kTestHandle2 = 456;
 static const uint16_t kTestUniverse = 456u;
+static const uint16_t kTestUniverse2 = 789u;
```

Comment:
Always default to constexpr
```cpp
constexpr uint16_t kTestUniverse2 = 789u;
```

Reply:
https://github.com/ETCLabs/sACN/pull/47/commits/afa59c256bbb4d724ec85fb29f0748d8eed55ed9